### PR TITLE
Add barcode detector frame on qr scan layout

### LINKED ISSUE #146: Add detector frame on QR scan layout
**UPDATED:**
Currently qr scan activity is just full screen activity. It gives no idea about what actually it is meant for.

Expected: 
- add detector frame on qr scan layout

I am working on it

### PR DESCRIPTION
**Pull request by mejariamol**
fixes #146 

screenshot_2017-06-02-09-17-50

### CODE REVIEW

**Comment by mejariamol** [comment]
@iamareebjamal please review

File: app/src/main/java/org/fossasia/openevent/app/qrscan/ScanQRActivity.java [hunk 1]
```diff
@@ -112,6 +135,66 @@ public void onRequestPermissionsResult(int requestCode, @NonNull String[] permis
 
     // Lifecycle methods end
 
+    private int convertDpToPixel(float dp) {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/qrscan/ScanQRActivity.java]: Move to View utils as static method

**Comment by mejariamol** [comment]
Actually we need a **static frame** on the layout for always.

**Comment by iamareebjamal** [comment]
The static frame like this implies that the camera will only scan codes inside this region. Which isn't true

**Comment by mejariamol** [comment]
yes. We will be allowing full screen scan detection.

**Comment by mejariamol** [comment]
>The static frame like this implies that the camera will only scan codes inside this region. Which isn't true

its not limiting detection area. But yes, capturing qr code in center is always advisable right?

**Comment by mejariamol** [comment]
@iamareebjamal please review

**Comment by iamareebjamal** [comment]
@mejariamol Please resolve conflicts

**Comment by mejariamol** [comment]
@iamareebjamal done :+1:

**Comment by iamareebjamal** [comment]
@mejariamol The current code is looking very hacky. Although I understand that we need this, do take a look at Vision APIs detectors and add that here instead because that was needed but was delegated for later. Once you add it, you can see for yourself if we need this. And if we do, you can create a custom view for camera frames which will be easy to make rather than setting 4 frames dynamically like this

**Comment by mejariamol** [comment]
>do take a look at Vision APIs detectors and add that here instead

Yeah. I will open new issue for that. Then we will decide if we need this. :+1: 
>you can create a custom view for camera frames which will be easy to make rather than setting 4 frames dynamically like this

I didn't understand custom view. I am setting up framelayouts dynamically to adjust with all screen.

**Comment by iamareebjamal** [comment]
@mejariamol True, but setting a customview would be better and more efficient than the current technique

File: app/src/main/res/layout/activity_scan_qr.xml [hunk 2]
```diff
@@ -20,6 +20,42 @@
...
+        android:layout_gravity="center"
+        android:layout_marginBottom="160dp"
+        android:layout_marginLeft="160dp"
+        android:background="@drawable/bordered_rectangle"/>
+
+    <FrameLayout
+        android:id="@+id/frame_corner_10"
+        android:layout_width="50dp"
+        android:layout_height="50dp"
+        android:layout_gravity="center"
```
Comment (arp95) [inline app/src/main/res/layout/activity_scan_qr.xml]: Nice work!
can you provide all these values in dimens.xml please.

**Comment by arp95** [comment]
@mejariamol please squash the commits.

**Comment by mejariamol** [comment]
@arp95 
>Its not working fluently

What's not working fluently?
btw we are not going for this. Instead google vision's barcode detector frame will be implemented. I am working on it.

**Comment by arp95** [comment]
@mejariamol sure go ahead.

File: app/src/main/java/org/fossasia/openevent/app/qrscan/ScanQRActivity.java [hunk 3]
```diff
@@ -5,14 +5,24 @@
...
+import android.graphics.drawable.LayerDrawable;
 import android.os.Bundle;
 import android.support.annotation.NonNull;
+import android.support.design.widget.CoordinatorLayout;
 import android.support.v4.app.ActivityCompat;
 import android.support.v4.content.ContextCompat;
+import android.support.v4.content.res.ResourcesCompat;
 import android.support.v7.app.AppCompatActivity;
+
+import android.util.DisplayMetrics;
```
Comment (rohanagarwal94) [inline app/src/main/java/org/fossasia/openevent/app/qrscan/ScanQRActivity.java]: Please avoid unused imports as mentioned in the codacy review.

Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/qrscan/ScanQRActivity.java]: @rohanagarwal94 The PR is not active as mentioned above

**Comment by mejariamol** [comment]
Will be handled in the new issue to add qr code detector frame provided by google vision api.

**Pull request closed without merging**